src/sprites.py

```diff
@@ -1,101 +1,158 @@
+#         self.rect = self.image.get_frect(center=pos)
 
+class AllSprites(pygame.sprite.Group):
+    '''A sprite group that handles every sprite and handles the camera logic'''
     def __init__(self):
+        super().__init__()
+
+        self.display_surface = pygame.display.get_surface()
+        if not self.display_surface:
+            raise ValueError("Display surface is not initialized")
+        self.offset = pygame.math.Vector2()
+
+    def draw(self, player_center):
+        self.offset.x = -(player_center[0] - SCREEN_WIDTH / 2)
+        self.offset.y = -(player_center[1] - SCREEN_HEIGHT / 2)
+
+        for sprite in self:
+            self.display_surface.blit(sprite.image, sprite.rect.topleft + self.offset)
+
+class Player(pygame.sprite.Sprite):
+    """move tile by tile"""
+
+    def __init__(self, pos, groups):
+        super().__init__(groups)
         # TODO: replace with actual images
-        self.image = pygame.Surface(size=(TILE_SIZE, TILE_SIZE))
-        self.image.fill("#ff0000")
+
+        self.image = pygame.Surface((TILE_SIZE, TILE_SIZE))
+        self.image.fill("red")
+        self.rect = self.image.get_frect(center=pos)
+        self.direction = pygame.math.Vector2()
+        self.speed = 250
+
         self.player_preview = self.image.copy()
         self.player_preview.set_alpha(128)
 
         self.inventory = Inventory()
 
-        self.rect: pygame.Rect = self.image.get_rect()
-        # keep track of the transparent preview of the next move
-        self.player_preview_rect = self.rect.copy()
-        # this is used to only move once when the mouse is pressed
         self.mouse_have_been_pressed: bool = False
-
-    def update(self) -> None:
-        """move the player and show a ghost to preview the move"""
-
-        # gost preview
-        mouse_pos = pygame.mouse.get_pos()
-
-        # get the relative pos of the player from the mouse
-        # to know on wich axis the player will move
-        delta_x = abs(self.rect.centerx - mouse_pos[0])
-        delta_y = abs(self.rect.centery - mouse_pos[1])
-
-        #  move the gost on the x axis
-        self.player_preview_rect = self.rect.copy()
-        if delta_x > delta_y:
-            if delta_x < (TILE_SIZE / 2):
-                # don't move the gost if the mouse is on the player hitbox
-                self.player_preview_rect.x = self.rect.x
-            elif mouse_pos[0] > self.rect.centerx:
-                # go right
-                self.player_preview_rect.x = self.rect.x + TILE_SIZE
-            else:
-                # go left
-                self.player_preview_rect.x = self.rect.x - TILE_SIZE
-        # move the gost on the y axis
+    #     self.rect: pygame.Rect = self.image.get_rect()
+    #     # keep track of the transparent preview of the next move
+    #     self.player_preview_rect = self.rect.copy()
+    #     # this is used to only move once when the mouse is pressed
+
+    '''DEBUGGING MOVEMENT METHOD'''
```

Comment:
I think for debugging purposes, this WASD movement implementation is fine and functional, especially for testing basic player movement logic. However, I’d like to highlight a few concerns:

1. **Mouse interaction in the game loop:**
- Mouse interaction is a key feature of the game, as it drives multiple actions (movement, interacting with objects)
- By removing mouse-based interactions and replacing them with WASD movement, we lose a significant part of the game's intended design.

2. **Ghost preview visualization:**
- The shadow opacity method (from @PurpleProg implementation) effectively acts as a **preview of where the player can go**. This visualization is crucial for providing feedback to the player before committing to a move.
- WASD movement does not provide any visualization, which makes it harder for players to understand their options or plan movements, especially if we do add a turn base system into the player movement.
- The ghost system is not just a visual feature but an integral part of the gameplay, aligning with strategies like games as Civilization or Unciv.
- Example of how it could look like, our game is grid based so I could not find the best representation: 
![unciv player movement](https://github.com/user-attachments/assets/e88eb684-5d9b-4225-87f0-d8c577ee979f)


3. **Representation of the gameplay:**
- While WASD movement may simplify debugging, it diverges significantly from the game's core mechanics as describes in the GGD.
- A ghost preview system reinforces the idea of a deliberate, turn-based approach, providing clarity and preventing accidental moves.

4. **Suggestion**:
- If the WASD movement is intended purely for debugging, consider isolating it behind a debug flag or toggle, making sure that the primary interaction system (mouse-based with ghost preview) remains intact and active during gameplay. 
- Maintaining the visualization aspect alongside debugging tools would better represent the intended gameplay experience.

In summary, while the WASD movement approach serves a purpose for debugging, it doesn’t align with the visualization-first approach of the game. Retaining the mouse interaction and ghost preview in the main loop is essential for representing the core mechanics and player experience.